### PR TITLE
[Synapse] Add support for workspace package

### PR DESCRIPTION
**Pull request by wonner**
**Description**

Add some new commands to manage workspace package in synapse workspace.

az synapse workspace-package upload/upload-batch/list/show/delete

Add parameters `--package-action` and `--package` in command `az synapse spark pool update` to support modify spark pool custom library.

**Testing Guide**

**History Notes**

[Synapse] `az synapse workspace-package`: Support workspace package CRUD
[Synapse] `az synapse spark pool update`: Support add or remove specific packages

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Synapse

File: src/azure-cli/azure/cli/command_modules/synapse/_help.py [hunk 1]
```diff
@@ -1562,6 +1566,59 @@
...
+helps['synapse workspace-package'] = """
+type: group
+short-summary: Manage Synapse's workspace packages.
+"""
+
+helps['synapse workspace-package upload'] = """
+type: command
+short-summary: Uploads a local workspace package file to an Azure Synapse workspace.
+examples:
+  - name: Uploads a local workspace package file to an Azure Synapse workspace.
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/synapse/_help.py:1578]:
```suggestion
short-summary: Upload a local workspace package file to an Azure Synapse workspace.
examples:
  - name: Upload a local workspace package file to an Azure Synapse workspace.
```

File: src/azure-cli/azure/cli/command_modules/synapse/_help.py [hunk 2]
```diff
@@ -1562,6 +1566,59 @@
...
+    text: |-
+        az synapse workspace-package upload --workspace-name testsynapseworkspace \\
+          --package C:/package.jar
+"""
+
+helps['synapse workspace-package upload-batch'] = """
+type: command
+short-summary: Uploads workspace package files from a local directory to an Azure Synapse workspace.
+examples:
+  - name: Uploads workspace package files from a local directory to an Azure Synapse workspace.
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/synapse/_help.py:1588]:
```suggestion
short-summary: Upload workspace package files from a local directory to an Azure Synapse workspace.
examples:
  - name: Upload workspace package files from a local directory to an Azure Synapse workspace.
```

File: src/azure-cli/azure/cli/command_modules/synapse/tests/latest/test_synapse_scenario.py [hunk 3]
```diff
@@ -1150,7 +1146,7 @@ def _create_workspace(self, *additional_create_params):
         })
 
         # Create adlsgen2
-        # self._create_storage_account()
+        self._create_storage_account()
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/synapse/tests/latest/test_synapse_scenario.py:1149]: Why create storage account again even if there's `StorageAccountPreparer` already?

File: src/azure-cli/azure/cli/command_modules/synapse/_params.py [hunk 4]
```diff
@@ -171,6 +173,11 @@ def load_arguments(self, _):
                    help='The library requirements file.')
         c.argument('force', arg_type=get_three_state_flag(), help='The flag of force operation.')
 
+        # Custom Libraries
+        c.argument('package_action', arg_group='Custom Libraries', arg_type=get_enum_type(['Add', 'Remove']),
+                   help='Package action must be specified when you add or remove a workspace package from a Apache Spark pool.')
+        c.argument('package', arg_group='Custom Libraries', nargs='+', help='List of workspace packages name.')
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/synapse/_params.py:179]: If this parameter is not required, you should use `nargs='*'` instead. Or else customer always need to specify `--package` for `az synapse spark pool update`

**Comment by evelyn-ys** [comment]
For complicated PR, please update **History Notes** part in PR description since PR title can't handle all for release notes. See [Format PR Description](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#format-pr-description)

File: src/azure-cli/azure/cli/command_modules/synapse/_params.py (hunk 4 above)
Comment (wonner) [reply src/azure-cli/azure/cli/command_modules/synapse/_params.py:179]: I think users do not need to specify everytime when they run cmd `az synapse spark pool update`?

Below cmd works well
`az synapse spark pool update --resource-group xx --workspace-name xx --name xx`

I only got error when
`az synapse spark pool update --resource-group xx--workspace-name xx --name xx --package-action Remove --package`
argument --package: expected at least one argument

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**